### DISCUSSION TITLE
Hi need help with an error i got with Zustand/React/Vite -> useEffect must not return anything besides a function

### DISCUSSION THREAD

**Discussion by eitai** [pmndrs/zustand#2644] [General] [answered]
Hey guys, I got an error 
useEffect must not return anything besides a function, which is used for clean-up. You returned: [object Object] at App .
when I try to set Data into a state in Zustand that's the error I get. any ideas?

![image](https://github.com/user-attachments/assets/5d3f9086-eb0c-47d2-b1ab-9eb7212841bb)
![image](https://github.com/user-attachments/assets/bf45a763-4f7d-44e9-8387-a23245dcec55)
![image](https://github.com/user-attachments/assets/89bc3b14-663e-4aac-9987-cb75a0b1cacc)

**Comment by dbritto-dev** [comment]
@eitai would you mind sharing us the whole component with the useEffect statement included?

**Reply by eitai** [reply]
Hi Dbritto, that's the thing, it's not the component it is complaining about the useEffect behind the scene of the Zustand.

**Reply by dbritto-dev** [reply, marked as answer]
@eitai in that case the error is not related to zustand but the code itself. As far I see in the compiled version you are returning an object in your `useEffect` statement which is wrong.

> **Note:** you can only return `functions` or `undefined` in `useEffect` statements

**Reply by eitai** [reply]
Yeah u right just got it solved , apparently in App I can return things in useEffect , thanks anyway :)
